backup/manager.go

```diff
@@ -241,16 +241,62 @@ func (b *Manager) IsSafeToRunNode(nodeID string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	for _, s := range snapshots {
 		if s.NodeID == nodeID && s.BackupID != "" && backupID != s.BackupID {
 			b.log.Errorf("remote restore was found for node %v.", nodeID)
 			b.log.Errorf("current backupID=%v, remote backupID-%v", backupID, s.BackupID)
 			return false, nil
+		} else if s.NodeID == nodeID {
+			b.log.Infof("Same nodeID, ensuring latest backup")
+			err := b.ensureLatestBackup(s)
+			if err != nil {
+				b.log.Errorf("ensureLatestBackup detected error: %v", err)
+				return false, err
+			}
 		}
 	}
+
 	return true, nil
 }
 
+// ensureLatestBackup checks whenever a given snapshot is older than our local version.
+// If the local version is older it proceeds to backup app data.
+func (b *Manager) ensureLatestBackup(snapshot SnapshotInfo) error {
+	latestBackup, err := b.db.latestBackupTime()
+	if err != nil {
+		return err
+	}
+	var layout string
+	if latestBackup == "" {
+		b.log.Infof("timestamp is empty, returning...")
+		return nil
+	}
+
+	// layout for snapshot.ModifiedTime.
+	layout = "2006-01-02 15:04:05 -0700"
+	// converting string to time.
+	appTimestamp, err := time.Parse(layout, latestBackup)
+	if err != nil {
+		b.log.Errorf("error parsing apptime exited with err: %w", err)
+		return err
+	}
+
+	// converting snapshots backup to time.
+	latestModifiedSnapshot, err := time.Parse(layout, snapshot.ModifiedTime)
+	if err != nil {
+		b.log.Errorf("error parsing latestModifiedSnapshot exited with err: %v", err)
+		return err
+	}
+
+	if appTimestamp.After(latestModifiedSnapshot) {
```

Comment:
Isn't it the other way around? Checking that the appTimestamp is before (older than) the provider one?

Reply:
No, this is correct but we can change the wording to make it more readable.

Reply:
The appTimestamp is the local one right? so if it is newer than the lastModifiedSnapshot (as the code states right now) we don't have a problem...

Reply:
Yes the `appTimestamp` is the local. So we only have a problem when the local is older than the lastModifiedSnapshot.  Where we wish to prompt the user to reinstall the app and restore from backup which I plan on adding functionality for.